contracts/EVMScriptFactories/UpdateTargetValidatorLimits.sol

```diff
@@ -12,7 +12,7 @@ import "../interfaces/INodeOperatorsRegistry.sol";
 contract UpdateTargetValidatorLimits is TrustedCaller, IEVMScriptFactory {
     struct TargetValidatorsLimit {
         uint256 nodeOperatorId;
-        bool isTargetLimitActive;
```

Comment:
There is currently no validation for the targetLimitMode value in the code, which could potentially lead to an unexpected dirty state in the Node Operators Registry (NOR). At present, targetLimitMode can only accept the values 0, 1, or 2. However, without proper validation, allowing other values could introduce unexpected behavior, especially if the mode is expanded in the future to support additional options. This lack of constraint could cause unforeseen issues

Reply:
From a smart contract perspective, there's no mistake here. You can specify a mode > 2 and it will be correctly handled by smart contract code when allocating new stake. By the current condition, any mode other than 0 includes limiting for stake allocation. 

The place in the NOR code: https://github.com/lidofinance/core/blob/737b675c6a7c6c9190a619bd001d1a34a3f28368/contracts/0.4.24/nos/NodeOperatorsRegistry.sol#L901

The difference between mode 1 and mode 2 (and possible modes > 2) now exists only in the offchain code.